### PR TITLE
Updates Hydroponics, Nerfs it, Adds Onions

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Ports over various hydroponics fixes updates, and balance alterations from TG.

- All hydroponics stats are limited at round-start, not just potency
  - You're going to have to get your machine upgraded if ya wanna get good stats now
- Ambrosia Gaia composting has been removed; you'll now have to grind up Gaia and dump earth's blood onto a tray to get it to become self-sustaining; 20 units per tray will be required
- Fixes splatting fruits not applying their reagents to a turf, just the turf's contents
- Weeds has been improved to starthistle (this is what it was anyway)
- Adds in regular and red onions
  - Slice them for onion rings; cook them for even more delicious onion rings!
- Adds reagent called concentrated onion juice--it's in red onions; it really makes your eyes water and fuss (temporary blindness and blurriness)
- Made glowshrooms and shadowshrooms less saturated
- Bonfires can be lit with slightly less oxygen

img

:cl: Fox McCloud
tweak: Botany nerfed; will need ambrosia gaia earth's blood to make a tray self-sustaining
tweak: Gene machine now requires heavy upgrading to make use of transferring high quality genes from one plant to another
add: Starthistle can now be grown
add: Adds in red and regular onions; slice them for onion rings; cook them for even more delicious onion rings
/:cl:

### CODE REVIEW

**Comment by Mitchs98** [comment]
Eh. Honestly everything but the machine nerf seems. The machine nerf is only going to make the job less enjoyable for botany and the chef as well. We all know science doesn't even upgrade the kitchen most rounds, why would they bother upgrading botany? Seems pointless.

File: code/modules/hydroponics/grown/misc.dm [hunk 1]
```diff
@@ -1,18 +1,32 @@
...
+	growthstages = 3
+	growing_icon = 'icons/obj/hydroponics/growing_flowers.dmi'
 	genes = list(/datum/plant_gene/trait/plant_type/weed_hardy)
+	mutatelist = list(/obj/item/seeds/harebell)
+
+/obj/item/seeds/starthistle/harvest(mob/user)
+	var/obj/machinery/hydroponics/parent = loc
+	var/seed_count = yield
+	if(prob(getYield() * 20))
+		seed_count++
```
Comment (Crazylemon64) [inline code/modules/hydroponics/grown/misc.dm]: Why not just have `seed_count` be `yield + 1`, instead of this redundant increment? Also, since it's only used in this frame, perhaps it should be defined inside the if statement block

**Comment by taukausanake** [comment]
I was just trying to understand what it means for the machines to get nerfed. Was looking through the code trying to see what was changed

File: code/modules/hydroponics/hydroponics.dm [hunk 2]
```diff
@@ -720,33 +733,7 @@
...
-			return
-		user.visible_message("<span class='notice'>[user] gently pulls open the soil for [O] and places it inside.</span>", "<span class='notice'>You tenderly root [O] into [src].</span>")
-		user.drop_item()
-		qdel(O)
-		visible_message("<span class='boldnotice'>[src] begins to glow with a beautiful light!</span>")
-		self_sustaining = TRUE
-		update_icon()
-
-	else if(istype(O, /obj/item/weapon/reagent_containers) )  // Syringe stuff (and other reagent containers now too)
+	if(istype(O, /obj/item/weapon/reagent_containers) )  // Syringe stuff (and other reagent containers now too)
```
Comment (Crazylemon64) [inline code/modules/hydroponics/hydroponics.dm]: I think you should remove that trailing space after the istype?

**Comment by Fox-McCloud** [comment]
It basically means that if you have, say, a high yield plant (say it produces 10), you will only be able to extra a yield of 4 from it, initially; once it's upgraded to tier 4 parts, then you'll be able to pull out the full 10.

Each stat functions like this now: potency, yield, endurance, production speed, lifespan, weed rate, and weed chance.

File: code/game/machinery/vending.dm [hunk 3]
```diff
@@ -977,12 +977,12 @@
...
-						/obj/item/seeds/lime = 3,/obj/item/seeds/orange = 3, /obj/item/seeds/peanuts = 3, /obj/item/seeds/potato = 3,/obj/item/seeds/poppy = 3,
+						/obj/item/seeds/lime = 3,/obj/item/seeds/onion = 3,/obj/item/seeds/orange = 3, /obj/item/seeds/peanuts = 3, /obj/item/seeds/potato = 3,/obj/item/seeds/poppy = 3,
 						/obj/item/seeds/pumpkin = 3,/obj/item/seeds/replicapod = 3,/obj/item/seeds/wheat/rice = 3,/obj/item/seeds/soya = 3,/obj/item/seeds/sunflower = 3,
 						/obj/item/seeds/tea = 3,/obj/item/seeds/tobacco = 3,/obj/item/seeds/tomato = 3,
 						/obj/item/seeds/tower = 3,/obj/item/seeds/watermelon = 3,/obj/item/seeds/wheat = 3,/obj/item/seeds/whitebeet = 3)
 	contraband = list(/obj/item/seeds/amanita = 2,/obj/item/seeds/glowshroom = 2,/obj/item/seeds/liberty = 2,/obj/item/seeds/nettle = 2,
-						/obj/item/seeds/plump = 2,/obj/item/seeds/reishi = 2,/obj/item/seeds/cannabis = 3, /obj/item/seeds/fungus = 3, /obj/item/seeds/random = 2)
+						/obj/item/seeds/plump = 2,/obj/item/seeds/reishi = 2,/obj/item/seeds/cannabis = 3,/obj/item/seeds/starthistle = 2,/obj/item/seeds/fungus = 3, /obj/item/seeds/random = 2)
```
Comment (Crazylemon64) [inline code/game/machinery/vending.dm]: the spacing here is really inconsistent

**Comment by taukausanake** [comment]
I don't know how I feel about that. On the one side it means that you need to upgrade or suffer, but on the other it _really_ limits your plants without those upgrades. At the same time though it does mean you need to be more careful with planting. Maybe I rather the machines really be for Xenobotany.

Now that I think about it, Hydro table component upgrades are already dead with Gaia around. No reason to upgrade the matter bins inside those at all. Why bother upgrading ANYTHING in Botany?

File: code/modules/food_and_drinks/recipes/recipes_microwave.dm [hunk 4]
```diff
@@ -541,6 +541,10 @@ datum/recipe/microwave/slimesandwich
 	being_cooked.reagents.del_reagent("toxin")
 	return being_cooked
 
+/datum/recipe/microwave/onionrings
```
Comment (FalseIncarnate) [inline code/modules/food_and_drinks/recipes/recipes_microwave.dm]: This could be a fryer special interaction instead of a microwave recipe.

Frankly, since it is literally a single ingredient that you are *probably* frying up (since onion rings are a deep-fried food), I'd argue this should move to the fryer as was done with French fries earlier this year.

**Comment by ktoma36** [comment]
Four is going to be the yield cap until upgrades? That limits a lot of things, generally making it more tedious to get to the fun parts of botany. I hope you haven't changed production rate as well, that would truly bury botany. Having crops the output low yields at a decreased rate would make botanist care for their plants more as well as make the botanist plan ahead, however it would greatly hinder the production of key plants that advanced botanist need to grow asap. The main ones I am thinking of are blumkins and gaia, both are essential for botany to expand and diversify. 

Another issue I find with this is that scientist will usually strait up refuse to upgrade botany, knowing that doing so will enable them to grow a bunch of meme plants with double the potency. Other times they cant be bothered to upgrade botany simply because the gene splicing machine is out of sight from outside botany. 

Here are a couple of ideas I have:

- Have the roundstart cap for yield be set for 6. A yield of 6 is decent at best. A yield of 4 is just to little, plus if the plant goes fowl while growing its just that much easier to get nothing from the plant (yield of >1). 

- Move the machines from the back of botany to the east wall and move those trays to the north wall. This will let scientist be able to easily upgrade the machines while walking.

- Decrease the material cost for the biomatter building thing (Cant remember its name, the thing you shove your spare plants into to turn to goods). With less plants being harvested overall it would make using the biomatter machine a resource sink. The chef needs meat and monkey cubes are expensive enough as is.

Overall my opinion of this PR is that it is turning botany away from genetically engineering super plants and more into just watching regular plants grow, _which is boring_. Making things more boring to the general population is a bad idea, we already have most botanist running around greytiding at midshift.

**Comment by uraniummeltdown** [comment]
Now that Botany is getting nerfed can you remove the time delay on growing plants that someone else added? Makes Botany artificially longer and boring

**Comment by Fox-McCloud** [comment]
@uraniummeltdown 

I was talking about that with @Crazylemon64 a couple hours ago, actually. If this does end up making botany a super drag/drudgery/takes too long, I told him that's the first thing I'm looking at.

**Comment by taukausanake** [comment]
I mean this isn't really a huge nerf as it sounds. Not for real botanists who like growing food. This just makes growing all the food a bit slower without upgrades. If Hydroponics didn't use the machines, if they were just about growing plants and not science-ing them then it wouldn't be an issue.
~Xenobotany when?~

**Comment by alexkar598** [comment]
Why exactly dose botany needs nerfing?

**Comment by FlattestGuitar** [comment]
I approve of onions, but I really don't see why lowering the stats of Botany machines is needed. The job itself is tedious enough as it is.

**Comment by adamkad1** [comment]
assuming this is in fact all liketg, you can NEVER pull out full, even with best parts, only like 95% or so

**Comment by Fullofskittles** [comment]
I dont really like the idea of having yet another job rely on science actually upgrading things to even be able to do a good chunk of their job. It's not just something that will cripple botanists if science refuses or neglects to upgrade their things at a reasonable time. The chef and thus the high population of hungry crew will potentially suffer as well if botany cannot produce enough things for the chef. It's difficult enough as a chef to get people to upgrade the kitchen...botanists are likely to be even lower on the totem pole of important things.

File: code/modules/hydroponics/gene_modder.dm [hunk 5]
```diff
@@ -40,13 +41,27 @@
...
-		if(istype(I, /obj/item/weapon/stock_parts))
-			var/obj/item/weapon/stock_parts/S = I
-			rating += S.rating-1
-	max_extract_pot = initial(max_extract_pot) + rating*5
+/obj/machinery/plantgenes/RefreshParts() // Comments represent the max you can set per tier, respectively. seeds.dm [219] clamps these for us but we don't want to mislead the viewer.
+	for(var/obj/item/weapon/stock_parts/manipulator/M in component_parts)
+		if(M.rating > 3)
+			max_potency = 95
+		else
+			max_potency = initial(max_potency) + (M.rating**3) // 53,59,77,95 	 Clamps at 100
```
Comment (MarsM0nd) [inline code/modules/hydroponics/gene_modder.dm]: Given the commet, it looks like you want 3^rating, but wouldn't that be rating^3 in this order?

**Comment by MarsM0nd** [comment]
I'm not sure if the limits on the first level should be like that. Given that potency of 50 does already help with quite some plants, but production of 12, which is the maximum production any plant has, which is a single mutated plant, and only a few more having production of 10, i would argue that production should start at 8. Still good for some plants to upgrade, while sure being better upgraded. 
For me, it seems like the changes would almost remove the DNA manipulator, as plants have better stats out of the tin, besides potency.

**Comment by Squirgenheimer** [comment]
I do admit it's quite understandable to be limiting the amount of production capability that botany has especially when you see people shocking groups of 3+ people over and over, essentially permastunning them from a distance. However,

If I'm looking at this right...
Given that the requirement per tray is 20 units of earthsblood, and potency is limited to 53 and yield to 4 this means that roundstart you need to harvest gaia about ~2 times in order to get a gaia tray. And the limit for production speed at roundstart appears to be 9. 

From testing 9 seems to about 6 minutes, so that's quite a long time in order to be able to put gaia in trays. Without upgrades you're looking at **12 minutes** _not counting maturation time_ for just one.
As such, production time limit should be **at least 6**, 12 is far too long a time and I haven't seen a roundstart seed with such a terrible stat before. It's also a really poor stat for the chefs, if I'm doing this right 3 harvests will only give the chef about **8 dough in 18 minutes** (again not counting maturation time, and assuming 50 potency gives 10 flour per wheat).  You _could_ have multiple plants or get upgrades at some point during that time, but it still seems pretty bad -- we should make the no-upgrades scenario more bearable than this.

**Comment by adamkad1** [comment]
just another reason to never branch out if chef does not ask, only ever use one maxed plant for all you need

**Comment by Jountax** [comment]
@uraniummeltdown There's a nerf to grow speed to plants added by other people? What?

@Squirgenheimer You can increase the flour output of wheat by giving it a better plant matter gene, as those don't seem to be limited by upgrades, and plant matter apparently transmutes into flour when a plant is blended as long as that plant is wheat.

File: code/modules/hydroponics/gene_modder.dm (hunk 5 above)
Comment (Crazylemon64) [inline code/modules/hydroponics/gene_modder.dm]: For `rating**3`

`1`: 51
`2`: 58
`3`: 77
`4`: 95

For `3**rating`:
`1`: 53
`2`: 59
`3`: 77
`4`: 95

Mars is correct

**Comment by uraniummeltdown** [comment]
@Jountax this PR https://github.com/ParadiseSS13/Paradise/pull/6430 made it so Paradise plants take 2x as long to grow as tg ones

**Comment by Jountax** [comment]
9 production speed out of the gate is...

It makes me very very sad.

**Comment by ktoma36** [comment]
Mmm... I love not being able to get the chef tomatos for 27 minutes.

**Comment by ktoma36** [comment]
All the stats are capped at such horrible amounts. Lifespan 35, weed growth 7, weed vulnerability 51, yield 4, endurance 35.

**Comment by alexkar598** [comment]
Why did botany need a nerf anyways?

**Comment by taukausanake** [comment]
@alexkar598 It didn't. It's just another parity port from /tg/ because that's what our server is based off of now

**Comment by ktoma36** [comment]
After a couple rounds of playing with this nerf I can officially say that it makes it impossible for a botanist to maintain over 8 trays alone _with_ every plant having weed adaption. This is also with moving trays around to reduce time needed to get to each of them.

**Comment by alexkar598** [comment]
Can we rename bonatist to grey tide with more access?

That's what botany is now with this nerf

**Comment by ktoma36** [comment]
This PR is an example of too many changes applied at once. The gaia nerf on its own might of been fine, the stats nerf on its own might of been fine. Both at once? I don't know how you are going to get usable data from this (if you are paying attention to the community backlash at all mind you) unless you revert one nerf and leave the other.

**Comment by taukausanake** [comment]
The gaia nerf is fine no matter what. I think it's fair by itself and maybe needed because it cuts out a lot of required work for Botany. They should work for the ability for less work, as all upgrades should be. The stats though is not as good however

**Comment by ktoma36** [comment]
So think a month is a good enough time to review the effects of this PR?

**Comment by Jountax** [comment]
I stand by my claim (made elsewhere) that for botanists who know what they're doing, the gaia nerf is actually a buff. Managed to get all the trays golden by 12:25 once, with other botanists focusing on actually taking care of the plants while I did so.

After learning about the effects of certain chemicals on plants as well, I'm less sad about the nerf to production speed genes, but it does revive my desire for xenobotany to be a thing so I can stop bugging the chemist every five seconds/bugging science for parts for a chem dispenser.

**Comment by Wh00pAss** [comment]
The gaia nerf is a complete waste of time. All is has done is take up more trays to get the stuff done as fast as possible, thus limiting the number of crops available for the chef. If you can RNG a nice wheat before the gaia grows then you can have amazing gaia production anyway. Cmplete waste of time other than flushing it out to be a little more involved.

As for the DNA machines, this is a total cluster truck. 80-90% of the shifts where I am in botany the machines are either never upgraded or never upgraded to max. I always move the machines to the front of botany to try and prompt RnD to come in and upgrade so I am never hiding at the back when/if they pass. The lack of upgrades makes a round start heavily dependent on chem for unstable or a delay in doing anything other than grinding the start of the shift until you have a good unstable wheat made. When you have that, you are at the mercy of the RNG gods getting something good out of the trays. I only use the DNA machines for perennial and weed adaptation now, that whole part of the department is dead. I've had it where I've got all requirements for max tier RnD and they just turn around and say "cant be bothered to do upgrades this shift"

In short rounds, botany cant contribute because of the huge grow times on plants due to the lack of modification from the DNA machine. I just sit there waiting for the round to end.

The department has gone from one that was fun where I could experiment with different dna strains in plants, to being a glorified food production line for the chef and maybe getting a couple of targeted plants out in a shift (if you are lucky). You literally cant get a decent dna plant as an antag now until RnD upgrades your machines... IF they upgrade them. There's no point teaching new players the ropes in the department (which I used to enjoy) as teaching them about DNA modification is totally at the mercy of RnD.
